.github/workflows/check.yml

```diff
@@ -141,23 +141,59 @@ jobs:
         if: matrix.rust-toolchain == 'stable'
         run: cargo semver-checks
 
-  check-freebsd:
+  check-vm:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [freebsd, openbsd, netbsd] # rust on solaris is too old
     runs-on: ubuntu-latest
+
     steps:
-      - name: Checkout
-        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
 
-      - name: Get "Install Rust" action from neqo
-        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - if: matrix.os == 'freebsd'
+        uses: vmactions/freebsd-vm@d7b8fcc7711aa41ad45e8d9b737cf90f035a7e3d
         with:
-          repository: mozilla/neqo
-          sparse-checkout: |
-            .github/actions/rust
-          path: neqo
-
-      - name: Install Rust
-        uses: ./neqo/.github/actions/rust
+          usesh: true
+          copyback: false
+          prepare: |
+            mkdir -p /usr/local/etc/pkg/repos
+            sed 's/quarterly/latest/' /etc/pkg/FreeBSD.conf > /usr/local/etc/pkg/repos/FreeBSD.conf
+            pkg update
+            pkg install -y rust
+          run: |
+            cargo check --all-targets
+            RUST_LOG=trace cargo test --no-fail-fast
+
+      - if: matrix.os == 'openbsd'
+        uses: vmactions/openbsd-vm@ebafa4eac4adf5e7d04e5bbb4aa764b75dd160df
         with:
-          targets: x86_64-unknown-freebsd
-
-      - run: cargo check --target x86_64-unknown-freebsd
+          usesh: true
+          copyback: false
+          prepare: |
+            pkg_add rust
+          run: |
+            cargo check --all-targets
+            RUST_LOG=trace cargo test --no-fail-fast
+
+      - if: matrix.os == 'netbsd'
+        uses: vmactions/netbsd-vm@dd0161ecbb6386e562fd098acf367633501487a4
+        with:
+          usesh: true
+          copyback: false
+          prepare: |
+            /usr/sbin/pkg_add rust
+          run: |
+            cargo check --all-targets
+            RUST_LOG=trace cargo test --no-fail-fast
+
+      - if: matrix.os == 'solaris'
+        uses: vmactions/solaris-vm@960d7483ffd6ac03397964cf6423a2f41332c9c8
+        with:
+          usesh: true
+          copyback: false
+          prepare: |
+            pkg install cargo
+          run: |
+            cargo check --all-targets
+            RUST_LOG=trace cargo test --no-fail-fast
```

Comment:
>  # rust on solaris is too old

Given this comment above, this step can be removed? Or do you expect we can introduce it in the near future?

```suggestion
```

Reply:
I hope we can add it soon-ish. I'd be nice to be able to support another Firefox tier-3 platform here.